core/src/main/scala/org/apache/spark/sql/delta/DeltaConfig.scala

```diff
@@ -145,32 +145,40 @@ trait DeltaConfigsBase extends DeltaLogging {
    * Validates specified configurations and returns the normalized key -> value map.
    */
   def validateConfigurations(configurations: Map[String, String]): Map[String, String] = {
-    configurations.map {
-      case kv @ (key, value) if key.toLowerCase(Locale.ROOT).startsWith("delta.constraints.") =>
-        // This is a CHECK constraint, we should allow it.
-        kv
-      case kv @ (key, value) if key.toLowerCase(Locale.ROOT).startsWith("delta.") =>
-        Option(entries.get(key.toLowerCase(Locale.ROOT).stripPrefix("delta."))) match {
-          case Some(deltaConfig) => deltaConfig(value) // validate the value
-          case None if
-            SparkSession.active.sessionState.conf
-              .getConf(DeltaSQLConf.ALLOW_ARBITRARY_TABLE_PROPERTIES)
+    val deltaLegazyOptions = SparkSession.active.sessionState.conf
+      .getConf(DeltaSQLConf.DELTA_LEGACY_STORE_WRITER_OPTIONS_AS_PROPS)
```

Comment:
this conf has a different meaning than we want here... I think we may need to create a new one for this change

Reply:
It makes sense, if its the only place affected, it's better to move it. I will make the changes

Reply:
I've changed where is applied the logic, created a config value to keep the previous logic, and adapted the test to check it all.